### PR TITLE
New metadata 1.0

### PR DESCRIPTION
**Pull request by abadger**
##### SUMMARY
A new metadata format was approved here: https://github.com/ansible/proposals/issues/54

* Update documentation to the new metadata format
* Changes to metadata-tool to account for new metadata  
    * Add GPL license header
    * Add upgrade subcommand to upgrade metadata version
    * Change default metadata to the new format
    * Fix exclusion of non-modules from the metadata report
* Update the metadata in all modules to the new version

##### ISSUE TYPE
 - Feature Pull Request
 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME
metadata of all modules.

##### ANSIBLE VERSION

```
2.3
```

### CODE REVIEW

**Review by dharmabumstead — APPROVED** [review]

File: docs/docsite/rst/dev_guide/developing_modules_checklist.rst [hunk 1]
```diff
@@ -31,9 +31,9 @@ The following  checklist items are important guidelines for people who want to c
 
     ANSIBLE_METADATA = {'status': ['preview'],
                         'supported_by': 'community',
-                        'version': '1.0'}
+                        'metadata_version': '1.0'}
 
-The complete module metadata specification is here: https://github.com/ansible/proposals/issues/30
+The complete module metadata specification is here: http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#ansible-metadata-block
```
Comment (mattclay) [inline docs/docsite/rst/dev_guide/developing_modules_checklist.rst]: Use `https` for docs links.

File: docs/docsite/rst/dev_guide/developing_modules_documenting.rst [hunk 2]
```diff
@@ -33,23 +33,28 @@ All modules must have the following sections defined in this order:
...
+   ANSIBLE_METADATA = {'metadata_version': '1.0',
+                       'status': ['preview'],
+                       'supported_by': 'community'}
 
-.. warning:: version field
+.. warning:: metadata_version field
 
-   This is the version of the ``ANSIBLE_METADATA`` schema, *not* the version of the module.
+   * This is the version of the ``ANSIBLE_METADATA`` schema, *not* the version of the module.
+   * Promoting a module's ``status`` or ``supported_by`` status should only be done by members of the Ansible Core Team.
```
Comment (mattclay) [inline docs/docsite/rst/dev_guide/developing_modules_documenting.rst]: The note about `status` and `supported_by` doesn't make sense under the "metadata_version field" heading. Perhaps the heading should just be a general warning instead?

**Review by mattclay — CHANGES_REQUESTED** [review]
A few minor docs changes requested.

**Comment by abadger** [comment]
Addressed mattclay's requests.  Waiting on shippable and then will merge.

**Review by mattclay — APPROVED** [review]

**Comment by gherrin** [comment]
To whom it may concern, this module misses views when granting table-level privs and/or ALL_IN_SCHEMA privs.  This section of code needs to select relkind of 'r' and 'v':

    def get_table_acls(self, schema, tables):
        query = """SELECT relacl
                   FROM pg_catalog.pg_class c
                   JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace
                   WHERE nspname = %s AND relkind = 'r' AND relname = ANY (%s)
                   ORDER BY relname"""
        self.cursor.execute(query, (schema, tables))
        return [t[0] for t in self.cursor.fetchall()]

**Pull request merged**